app/infra/notificationsService/notificationsService.js

```diff
@@ -0,0 +1,42 @@
+import path from 'path';
+import { ipcRenderer } from 'electron';
+import { ipcConsts } from '/vars';
+
+// @flow
+class NotificationsService {
+  static notify = ({ title, notification, callback }: { title: string, notification: string, callback: ({ isClicked: boolean }) => void }) => {
+    NotificationsService.isNotificationPermitted().then(({ isNotificationAllowed }: { isNotificationAllowed: boolean }) => {
+      if (isNotificationAllowed) {
+        const notificationOptions: any = {
+          body: notification,
+          icon: path.join(__dirname, '..', 'app', 'assets', 'icons', 'app_icon.png')
+        };
+        const desktopNotification = new Notification(title || 'Alert', notificationOptions);
+        desktopNotification.onclick = () => {
+          ipcRenderer.send(ipcConsts.NOTIFICATION_CLICK);
+          ipcRenderer.once(ipcConsts.NOTIFICATION_CLICK_SUCCESS, () => {
```

Comment:
there is no need to wait for the response, callback can be called immediately after calling show and focus
also callback doesn't do anything with passed param { isClicked: true } and it's useless

Reply:
ok